src/intermediate/IntermediateInstruction.h

```diff
@@ -125,6 +126,12 @@ namespace vc4c
 			SetFlag setFlags;
 			InstructionDecorations decoration;
 			bool canBeCombined;
+
+			/* TODO should be fix for lower cost implementation */
+			bool operator<(const IntermediateInstruction &il) const {
```

Comment:
Where do you need to order instructions? Usually, they are kept as pointer-to-instructions, which are sorted implicitly.

If you need to order them, you can order them by their pointers (e.g. `return this < *il`), since any other values are not distinct (e.g. all mutex-locks have the same `to_string()` value), and their pointers do not change, since the objects are not moved.

Reply:
This is my mistake. Removed.